scaffolding-python/lib/scaffolding.sh

```diff
@@ -0,0 +1,78 @@
+#
+# DO NOT OVERRIDE `do_default_xxxxx` IN PLANS!
+#
+
+#
+# Internal functions. Do not use these in your plans as they can change.
+#
+
+_example() {
+  return 127
+}
+
+#
+# Scaffolding Variables
+#
+
+scaffolding_python_pkg="core/python"
+
+#
+# Scaffolding Callback Functions
+#
+
+# The default_begin phase is executed prior to loading the scaffolding. As such
+# we have a scaffolding specific callback to allow us to run anything we need
+# to execute before download and build. This callback executes immediately
+# after the scaffolding is loaded.
+_scaffolding_begin() {
+  pkg_deps=(${pkg_deps[@]} $scaffolding_python_pkg)
+}
+
+
+# Recursively clean GOPATH and dependencies.
+do_default_clean() {
+  scaffolding_python_clean
+}
+
+scaffolding_python_clean() {
+  pushd "$pkg_dirname"
+  if [[ -f "setup.py" ]]; then
+    python setup.py clean --all
+  else
+    find . \( -type f -name "*.pyc" -or -name "*.pyo" \) -delete
+  fi
+  popd
+}
+
+scaffolding_python_build() {
+  python setup.py build
+}
+
+scaffolding_python_install() {
+  if [[ -f "${pkg_dirname}/setup.py" ]]; then
+    python setup.py install \
+      --prefix="$pkg_prefix"
```

Comment:
For now you can either test for `setuptools` using a command specific to it such as `python setup.py bdist_egg --help` and use the status code.

```bash
  python setup.py install \
    --prefix="$pkg_prefix" \
    --old-and-unmanageable || \
  python setup.py install \
    --prefix="$pkg_prefix"
```